### PR TITLE
[AOT Refactor] subclass utils

### PR DESCRIPTION
**Pull request by jon-chuang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114643
* #114564
* #114563
* #114562
* #114561
* #114559
* #114558
* #114557
* #114556
* #114555
* __->__ #114554
* #114553
* #114552
* #114551
* #114550
---

Part _ of https://github.com/pytorch/pytorch/issues/114548

### CODE REVIEW

**Review by bdhirsh — APPROVED** [review]

File: torch/_functorch/_aot_autograd/subclass_utils.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+from typing import Any, List, Optional, Tuple, Union
+
+import torch.utils._pytree as pytree
+
+from torch import Tensor
+from torch.utils._python_dispatch import is_traceable_wrapper_subclass
+
+from .schemas import SubclassCreationMeta, ViewAndMutationMeta
+
```
[the remark this answers is not in the document]
Comment (jon-chuang) [reply torch/_functorch/_aot_autograd/subclass_utils.py:9]: Actually, reading https://peps.python.org/pep-0257/, it seems that the recommended thing is to use """ comments at the beginning of the module. Let me do that.

**Pull request closed without merging**